Flaky script test test_wait_template_with_utcnow

<!-- READ THIS FIRST:
  - If you need additional help with this template, please refer to https://www.home-assistant.io/help/reporting_issues/
  - Make sure you are running the latest version of Home Assistant before reporting an issue: https://github.com/home-assistant/core/releases
  - Do not report issues for integrations if you are using custom components or integrations.
  - Provide as many details as possible. Paste logs, configuration samples and code into the backticks.
  DO NOT DELETE ANY TEXT from this template! Otherwise, your issue may be closed without comment.
-->
## The problem
<!-- 
  Describe the issue you are experiencing here to communicate to the
  maintainers. Tell us what you were trying to do and what happened.
-->
- `test_wait_template_with_utcnow` is flaky and fails regularly on CI.
- The test has been flaky for over a week at least.

## Environment
<!--
  Provide details about the versions you are using, which helps us to reproduce
  and find the issue quicker. Version information is found in the
  Home Assistant frontend: Configuration -> Info.
-->

- Home Assistant Core release with the issue: 2021.2.0.dev0
- Last working Home Assistant Core release (if known): 
- Operating environment (OS/Container/Supervised/Core): Core
- Integration causing this issue: helpers.script
- Link to integration documentation on our website: 

## Problem-relevant `configuration.yaml`
<!--
  An example configuration that caused the problem for you. Fill this out even
  if it seems unimportant to you. Please be sure to remove personal information
  like passwords, private URLs and other credentials.
-->

```yaml

```

## Traceback/Error logs
<!--
  If you come across any trace or error logs, please provide them.
-->

```txt
Error: test_wait_template_with_utcnow[pyloop]

asyncio.exceptions.TimeoutError
...................F............................................... [2615/3063]
................................................................... [2682/3063]
................................................................... [2749/3063]
................................................................... [2816/3063]
................................................................... [2884/3063]
................................................................... [2951/3063]
................................................................... [3018/3063]
--- Logging error ---
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/logging/__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 1744, in call_exception_handler
    self._exception_handler(self, context)
  File "/__w/core/core/homeassistant/runner.py", line 93, in _async_loop_exception_handler
    logging.getLogger(__package__).error(
Message: 'Error doing job: %s'
Arguments: ('Future exception was never retrieved',)
Error doing job: Future exception was never retrieved
aioambient.errors.AmbientError
.............................................                       [3063/3063]
=================================== FAILURES ===================================
____________________ test_wait_template_with_utcnow[pyloop] ____________________
[gw0] linux -- Python 3.8.2 /__w/core/core/venv/bin/python

hass = <homeassistant.core.HomeAssistant object at 0x7f5686e581c0>

    async def test_wait_template_with_utcnow(hass):
        """Test the wait template with utcnow."""
        sequence = cv.SCRIPT_SCHEMA({"wait_template": "{{ utcnow().hour == 12 }}"})
        script_obj = script.Script(hass, sequence, "Test Name", "test_domain")
        wait_started_flag = async_watch_for_action(script_obj, "wait")
        start_time = dt_util.utcnow().replace(minute=1) + timedelta(hours=48)
    
        try:
            hass.async_create_task(script_obj.async_run(context=Context()))
>           await asyncio.wait_for(wait_started_flag.wait(), 1)

tests/helpers/test_script.py:828: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

fut = <Task cancelled name='Task-39089' coro=<Event.wait() done, defined at /usr/local/lib/python3.8/asyncio/locks.py:296>>
timeout = 1

    async def wait_for(fut, timeout, *, loop=None):
        """Wait for the single Future or coroutine to complete, with timeout.
    
        Coroutine will be wrapped in Task.
    
        Returns result of the Future or coroutine.  When a timeout occurs,
        it cancels the task and raises TimeoutError.  To avoid the task
        cancellation, wrap it in shield().
    
        If the wait is cancelled, the task is also cancelled.
    
        This function is a coroutine.
        """
        if loop is None:
            loop = events.get_running_loop()
        else:
            warnings.warn("The loop argument is deprecated since Python 3.8, "
                          "and scheduled for removal in Python 3.10.",
                          DeprecationWarning, stacklevel=2)
    
        if timeout is None:
            return await fut
    
        if timeout <= 0:
            fut = ensure_future(fut, loop=loop)
    
            if fut.done():
                return fut.result()
    
            fut.cancel()
            raise exceptions.TimeoutError()
    
        waiter = loop.create_future()
        timeout_handle = loop.call_later(timeout, _release_waiter, waiter)
        cb = functools.partial(_release_waiter, waiter)
    
        fut = ensure_future(fut, loop=loop)
        fut.add_done_callback(cb)
    
        try:
            # wait until the future completes or the timeout
            try:
                await waiter
            except exceptions.CancelledError:
                fut.remove_done_callback(cb)
                fut.cancel()
                raise
    
            if fut.done():
                return fut.result()
            else:
                fut.remove_done_callback(cb)
                # We must ensure that the task is not running
                # after wait_for() returns.
                # See bugs.python.org/issue32751
                await _cancel_and_wait(fut, loop=loop)
>               raise exceptions.TimeoutError()
E               asyncio.exceptions.TimeoutError

/usr/local/lib/python3.8/asyncio/tasks.py:490: TimeoutError
---------------------------- Captured stderr setup -----------------------------
DEBUG:asyncio:Using selector: EpollSelector
------------------------------ Captured log setup ------------------------------
DEBUG    asyncio:selector_events.py:59 Using selector: EpollSelector
----------------------------- Captured stderr call -----------------------------
INFO:homeassistant.helpers.script.test_name:Test Name: Running test_domain script
INFO:homeassistant.helpers.script.test_name:Test Name: Executing step wait template
------------------------------ Captured log call -------------------------------
INFO     homeassistant.helpers.script.test_name:script.py:1138 Test Name: Running test_domain script
INFO     homeassistant.helpers.script.test_name:script.py:1138 Test Name: Executing step wait template
--------------------------- Captured stderr teardown ---------------------------
DEBUG:homeassistant.core:Bus:Handling <Event homeassistant_stop[L]>
DEBUG:homeassistant.core:Bus:Handling <Event homeassistant_final_write[L]>
DEBUG:homeassistant.core:Bus:Handling <Event homeassistant_close[L]>
---------------------------- Captured log teardown -----------------------------
DEBUG    homeassistant.core:core.py:704 Bus:Handling <Event homeassistant_stop[L]>
DEBUG    homeassistant.core:core.py:704 Bus:Handling <Event homeassistant_final_write[L]>
DEBUG    homeassistant.core:core.py:704 Bus:Handling <Event homeassistant_close[L]>
```

## Additional information

--- a/tests/helpers/test_script.py
+++ b/tests/helpers/test_script.py
@@ -824,9 +824,11 @@ async def test_wait_template_with_utcnow(hass):
     start_time = dt_util.utcnow().replace(minute=1) + timedelta(hours=48)
 
     try:
-        hass.async_create_task(script_obj.async_run(context=Context()))
-        await asyncio.wait_for(wait_started_flag.wait(), 1)
-        assert script_obj.is_running
+        non_maching_time = start_time.replace(hour=3)
+        with patch("homeassistant.util.dt.utcnow", return_value=non_maching_time):
+            hass.async_create_task(script_obj.async_run(context=Context()))
+            await asyncio.wait_for(wait_started_flag.wait(), 1)
+            assert script_obj.is_running
 
         match_time = start_time.replace(hour=12)
         with patch("homeassistant.util.dt.utcnow", return_value=match_time):
@@ -848,13 +850,17 @@ async def test_wait_template_with_utcnow_no_match(hass):
     timed_out = False
 
     try:
-        hass.async_create_task(script_obj.async_run(context=Context()))
-        await asyncio.wait_for(wait_started_flag.wait(), 1)
-        assert script_obj.is_running
-
         non_maching_time = start_time.replace(hour=3)
         with patch("homeassistant.util.dt.utcnow", return_value=non_maching_time):
-            async_fire_time_changed(hass, non_maching_time)
+            hass.async_create_task(script_obj.async_run(context=Context()))
+            await asyncio.wait_for(wait_started_flag.wait(), 1)
+            assert script_obj.is_running
+
+        second_non_maching_time = start_time.replace(hour=4)
+        with patch(
+            "homeassistant.util.dt.utcnow", return_value=second_non_maching_time
+        ):
+            async_fire_time_changed(hass, second_non_maching_time)
 
         with timeout(0.1):
             await hass.async_block_till_done()

